src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -2205,6 +2206,94 @@ public static bool FoundInUnboundLambda(BoundNode methodBody, IdentifierNameSynt
                 return base.Visit(node);
             }
         }
+
+        private sealed class WasPropertyBackingFieldAccessChecked : BoundTreeWalkerWithStackGuardWithoutRecursionOnTheLeftOfBinaryOperator
```

Comment:
It looks like it would also be reasonable to wrap this type declaration in `#if DEBUG`.